File: Dataset/Leetcode/train/8/601.py
 class Solution:
    def XXX(self, s: str) -> int:
        if len(s)==0:
            return 0
        i=0
        while s[i]==" ":
            i+=1
            if i==len(s):
                return 0
        s=s[i:]
        if s[0]!= '-' and s[0]!='+' and (s[0]<'0' or s[0]>'9'):
            return 0
        else:
            if len(s)==1 and s[0]!='-' and s[0]!='+':
                return int(s)
            elif len(s)==1 and (s[0]<'0' or s[0]>'9'):
                return 0
            i=1
            while s[i]>='0' and s[i]<='9':
                i+=1
                if i==len(s):
                    break
            if i==1 and (s[0]=='-' or s[0]=='+'):
                return 0
            else:
                num=int(s[0:i])
                if num>2**31-1:
                    return 2**31-1
                elif num<-2**31:
                    return -2**31
                else:
                    return num
        return 0

File: Dataset/Leetcode/test/28/601.java
 class Solution {
    public int XXX(String haystack, String needle) {
        int len = needle.length(),haylen = haystack.length();
        if(len==0) return 0;
        for(int i=0;i<haylen;i++){
            if(haystack.charAt(i)==needle.charAt(0)){
                if(i+len>haylen) return -1;  //后面的长度小于len,不足以进行切分
                String str = haystack.substring(i,i+len);
                if(str.equals(needle)) return i;
            }
        }
        return -1;
    }
}

